sockeye/inference.py

```diff
@@ -1363,6 +1369,9 @@ def _beam_search(self,
         lengths = mx.nd.ones((self.batch_size * self.beam_size, 1), ctx=self.context)
```

Comment:
That would cause the following error:
mxnet.base.MXNetError: Error in operator normalizefinishedhypotheses13__div0: [13:00:12] src/operator/contrib/../elemwise_op_common.h:123: Check failed: assign(&dattr, (*vec)[i]) Incompatible attr in node normalizefinishedhypotheses13__div0 at 1-th input: expected float32, got int32

I believe L1725: scores_accumulated / self.length_penalty(lengths),

Reply:
either way there has to be at least one casting because lengths is used as int type and as float type when computing the length penalty.